src/pulse/types.rs

```diff
@@ -57,22 +57,14 @@ pub mod cb {
 /// For types we only have pointers to. Use structs with names so there is at
 /// least pointer type safety.
 pub mod opaque {
-    #[repr(C)]
-    pub struct pa_mainloop;
-    #[repr(C)]
-    pub struct pa_mainloop_api;
-    #[repr(C)]
-    pub struct pa_context;
-    #[repr(C)]
-    pub struct pa_spawn_api;
-    #[repr(C)]
-    pub struct pa_threaded_mainloop;
-    #[repr(C)]
-    pub struct pa_proplist;
-    #[repr(C)]
-    pub struct pa_stream;
-    #[repr(C)]
-    pub struct pa_operation;
+    pub enum pa_mainloop {}
```

Comment:
Yup, changed this to work around the improper ctypes warning.

I had the same reservations as you, so I tried to follow some of the discussion around this. The Rust book recommends [using an empty enum](https://doc.rust-lang.org/book/ffi.html#representing-opaque-structs) (updated ~August: https://github.com/rust-lang/rust/pull/27542).